### DISCUSSION TITLE
How to contribute to example library?

### DISCUSSION THREAD

**Discussion by duarte-jfs** [HelgeGehring/femwell#160] [General]
Hi, I have used FEMWELL for RF transmission line design and have written a tutorial benchmarking the software for a simple CPW with some experimental results and I'd be happy to share it, but I'm a bit lost on how to contribute. Can anyone point me in the right direction?

**Comment by HelgeGehring** [comment]
Wow, that sounds amazing!

All the code for the examples you find in https://github.com/HelgeGehring/femwell/tree/main/docs

The electronics examples we collected in https://github.com/HelgeGehring/femwell/tree/main/docs/electronics/examples
If you experimental data is in an extra file, just make a folder.

In that folder every python file is an example.
The header
```
# ---
# jupyter:
#   jupytext:
#     formats: py:percent,md:myst
#     text_representation:
#       extension: .py
#       format_name: percent
#       format_version: '1.3'
#       jupytext_version: 1.15.0
#   kernelspec:
#     display_name: Python 3
#     name: python3
# ---
```
you can just copy from another example :D just make sure that is has the right language (python or julia, but if you copy from the same folder it shouldn't be a problem; in case you use jupyter notebooks you can convert them to such .py files via jupytext which allows to have them nicely in git)

The first headline defines the name of the example, for example:
```
# %% [markdown]
# # Coax cable
```

After that you use for explanations
```
# %% [markdown]
# text to explain stuff
# more text
```

and
```
# %%
plt.plot(...)
plt.show()
```
for code

you can hide cells for example by
```
# %% tags = ["hide-input"]
```
where the code needs to be opened with a click (usually i hide the imports in the beginning like that)
and you can do the same via "hide-output" for the output (in case you have some extra plots/prints which are nice for better understanding but would somewhat bloat the example)

The final step is to add the example to `https://github.com/HelgeGehring/femwell/blob/main/docs/_toc.yml`. You just need there a line with the path to the file. I'd guess the perfect spot would be after the coax cable?
